tests/frontend/specs/timeslider_follow.js

```diff
@@ -1,55 +1,118 @@
+  xit("only to lines that exist in the current pad view, see #4389", async function(){
+    // Select everything and clear via delete key
+    let e = helper.padInner$.Event(helper.evtType);
+    e.keyCode = 8; //delete key
+    let lines = helper.linesDiv();
+    helper.selectLines(lines[0], lines[lines.length - 1]); // select all lines
+    // probably unnecessary, but wait for the selection to be Range not Caret
+    await helper.waitForPromise(function(){
+      return helper.padInner$.document.getSelection().type === 'Range';
+    })
+    helper.padInner$('#innerdocbody').trigger(e);
+    await helper.waitForPromise(function(){
+      return helper.commits.length === 1; 
+    })
+    await helper.edit("Test line\n\n")
+    await helper.edit("Another test line", 3)
 
-        setTimeout(function() {
-          //make sure the text has changed
-          var newTop = timeslider$('#innerdocbody').offset();
-          expect( originalTop ).not.to.eql( newTop );
-          done();
-        }, 1000);
+    await helper.gotoTimeslider();
 
-      }, 2000);
-    }, revs*timePerRev);
-  });
+    // set to follow contents as it arrives
+    helper.contentWindow().$('#options-followContents').prop("checked", true);
+
+    let oldYPosition = helper.contentWindow().$("#editorcontainerbox")[0].scrollTop;
+    expect(oldYPosition).to.be(0);
+
+    /**
+     * pad content rev 0 [default Pad text]
+     * pad content rev 1 ['']
+     * pad content rev 2 ['Test line','','']
+     * pad content rev 3 ['Test line','','Another test line']
+     */
+
+    // line 3 changed
+    helper.contentWindow().$('#leftstep').click();
+    await helper.waitForPromise(function(){
+      return hasFollowedToLine(3);
+    })
+
+    // line 1 is the first line that changed
+    helper.contentWindow().$('#leftstep').click();
+    await helper.waitForPromise(function(){
+      return hasFollowedToLine(1);
+    })
 
+    // line 1 changed
+    helper.contentWindow().$('#leftstep').click();
+    await helper.waitForPromise(function(){
+      return hasFollowedToLine(1);
+    })
+
+    // line 1 changed
+    helper.contentWindow().$('#rightstep').click();
+    await helper.waitForPromise(function(){
+      return hasFollowedToLine(1);
+    })
+
+    // line 1 is the first line that changed
+    helper.contentWindow().$('#rightstep').click();
+    await helper.waitForPromise(function(){
+      return hasFollowedToLine(1);
+    })
+
+    // line 3 changed
+    helper.contentWindow().$('#rightstep').click();
+    return helper.waitForPromise(function(){
+      return hasFollowedToLine(3);
+    })
+  })
 });
 
+/**
+ * @param {number} lineNum 
+ * @returns {boolean} scrolled to the lineOffset?
+ */
+function hasFollowedToLine(lineNum) {
+  let scrollPosition = helper.contentWindow().$("#editorcontainerbox")[0].scrollTop;
+  let lineOffset = helper.contentWindow().$('#innerdocbody').find(`div:nth-child(${lineNum})`)[0].offsetTop;
+
+  return Math.abs(scrollPosition - lineOffset) < 1;
```

Comment:
I think this will only be true if the number of lines of content in the pad is at least `lineNum` more than can be displayed in the visible area at once. The test is only using 3 lines of content, which is probably too few to make a scrollbar appear let alone be able to scroll down to the top of the 3rd line.

Reply:
Oh. I always have the scrollbar displayed when in timeslider. Can you verify? (tested on android/chromium/firefox)

Reply:
I don't understand CSS (or whatever makes this possible) enough to be able to find the reason why, but the scrollbar always has some "space"/"room" downwards. so for me it can be scrolled even on pads with no/few lines.

Is there something special to my setup?!

Reply:
It looks like the extra blank room at the bottom is equal to the height of the timeslider toolbar. I also don't know enough about CSS to know why that is. As long as the offset to the top of the third line is less than the height of the timeslider toolbar then I think it'll work. However, the test might be fragile and hard to fix if it depends on the height of the toolbar. I'm not sure how to make it better.